src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/membership_table.go

```diff
@@ -41,14 +41,11 @@ CREATE UNIQUE INDEX IF NOT EXISTS account_membership_event_id ON account_members
 
 const insertMembershipSQL = `
 	INSERT INTO account_memberships(localpart, room_id, event_id) VALUES ($1, $2, $3)
-	ON CONFLICT (localpart, room_id) DO UPDATE SET event_id = EXCLUDED.event_id
+	ON CONFLICT (localpart, room_id) DO	UPDATE SET event_id = EXCLUDED.event_id
 `
 
-const selectMembershipSQL = "" +
-	"SELECT * from account_memberships WHERE localpart = $1 AND room_id = $2"
-
 const selectMembershipsByLocalpartSQL = "" +
-	"SELECT room_id, event_id FROM account_memberships WHERE localpart = $1"
+	"SELECT * FROM account_memberships WHERE localpart = $1"
```

Comment:
I'd prefer that we were explicit about the fields we are returning. Otherwise it will break in odd ways if we add a new column.